File: 2022/Day_1/solution.py
import time 
from typing import List

def main():
    file = open("input", "r")
    contents: List[str] = file.read().split('\n')

    # We could use tuple instead, but it will not
    # be a fair comparison. Not sure if tuple
    # works the same way as list in Python
    # behind the scenes.
    result: List[int] = []

    t0 = time.time_ns()
    result = solution(contents)
    t1 = time.time_ns()  
    print("Took: ", int((t1-t0)/1000)," microsecs")

    print("Elf ", result[0], " has the highest calories: ", result[1])

    
def solution(input: List[str]) -> List[int]:
    current_elf: int = 0
    sum_calories: int = 0
    highest_elf: int = 0
    highest_calories: int = 0

    for calorie in input:
        if (not calorie):
            if (sum_calories > highest_calories):
                highest_calories = sum_calories
                highest_elf = current_elf
            current_elf += 1
            sum_calories = 0
            continue
        sum_calories += int(calorie)

    return [highest_elf, highest_calories]
        
if __name__=="__main__":
    main()

File: 2022/Day_1/solution.ts
import { readFileSync } from "fs";

// Don't hate me for using main(), I'm used to Dart.
async function main(): Promise<void> {

    const contents = readFileSync('input').toString().split('\n');

    let result: [number, number] = [0,0];

    let t0: number = performance.now();
    result = solution(contents);
    let t1: number = performance.now();
    console.log(`Took: ${((t1-t0)*1000).toFixed(0)} microsecs`);
    
    console.log(`Elf ${result[0]} has the highest calories: ${result[1]}`);
};

const solution = (input: string[]): [number, number] => {

    let elfNumber: number = 0;
    let sumCalories: number = 0;
    let highestElf: number = 0;
    let highestCalories: number = 0;

    for (let calories of input) {
        if (!calories) {
            if (sumCalories > highestCalories) {
                highestCalories = sumCalories;
                highestElf = elfNumber;
            }
            elfNumber++;
            sumCalories = 0;
            continue;
        }

        sumCalories += Number.parseInt(calories);
    }

    return [highestElf, highestCalories];

};

main();
